hugegraph-core/src/main/java/com/baidu/hugegraph/backend/store/raft/RaftNode.java

```diff
@@ -59,61 +58,71 @@
 
     private static final Logger LOG = Log.logger(RaftNode.class);
 
-    private final String group;
-    private final StoreStateMachine stateMachine;
+    private final RaftSharedContext context;
     private final Node node;
+    private final StoreStateMachine stateMachine;
+    private final AtomicLong leaderTerm;
 
     private final Object electedLock;
     private volatile boolean elected;
     private volatile boolean started;
     private final AtomicInteger busyCounter;
 
-    public RaftNode(String group, BackendStore store,
-                    RaftSharedContext context) {
-        this.group = group;
-        this.stateMachine = new StoreStateMachine(store, this, context);
+    public RaftNode(RaftSharedContext context) {
+        this.context = context;
+        this.stateMachine = new StoreStateMachine(context);
         try {
-            this.node = this.initRaftNode(store, context);
+            this.node = this.initRaftNode();
         } catch (IOException e) {
             throw new BackendException("Failed to init raft node", e);
         }
         this.node.addReplicatorStateListener(new RaftNodeStateListener());
+        this.leaderTerm = new AtomicLong(-1);
         this.electedLock = new Object();
         this.elected = false;
         this.started = false;
         this.busyCounter = new AtomicInteger();
     }
 
-    public String group() {
-        return this.group;
+    public RaftSharedContext context() {
+        return this.context;
     }
 
     public Node node() {
         return this.node;
     }
 
-    public NodeId nodeId() {
-        return this.node.getNodeId();
+    public PeerId nodeId() {
+        return this.node.getNodeId().getPeerId();
+    }
+
+    public PeerId leaderId() {
+        return this.node.getLeaderId();
+    }
+
+    public boolean isRaftLeader() {
+        return this.leaderTerm.get() > 0;
+    }
+
+    public void leaderTerm(long term) {
+        this.leaderTerm.set(term);
     }
 
     public void shutdown() {
         this.node.shutdown();
     }
 
-    private Node initRaftNode(BackendStore store, RaftSharedContext context)
-                              throws IOException {
-        String storePath = store.database() + "-" + store.store();
-        NodeOptions nodeOptions = context.nodeOptions(storePath);
+    private Node initRaftNode() throws IOException {
+        NodeOptions nodeOptions = this.context.nodeOptions();
         nodeOptions.setFsm(this.stateMachine);
 
-        // TODO: When support sharding, groupId needs to be bound to shard Id
-        String groupId = storePath;
-        PeerId endpoint = context.endpoint();
-
+        String groupId = this.context.group();
```

Comment:
should keep comment "// TODO: When support sharding, groupId needs to be bound to shard Id"?